R/helpers.R

```diff
@@ -42,6 +42,31 @@ Validate <- setRefClass(
   )
 )
 
+# positive_number-class ----
+
+#' `positive_number`
+#'
+#' @description `r lifecycle::badge("experimental")`
+#'
+#' The [`positive_number`] class is a class to store not `NULL`, non `NA`,
+#' finite and strictly positive numerical value. It is mainly used to store
+#' reference dose value in model classes.
+#'
+#' @name positive_number
+#'
+positive_number <- setClass(
+  Class = "positive_number",
+  contains = "numeric",
+  validity = function(object) {
+    v <- Validate()
+    v$check(
+      test_number(object, lower = .Machine$double.xmin, finite = TRUE),
```

Comment:
actually how about simplifying this:
```suggestion
      test_number(object, finite = TRUE) && object > 0,
```
seems more explicit and does the same

Reply:
Sure, we can do that in this case as `a && b`, `b` is not evaluated if `a` `FALSE`, so we are sure that `object > 0` does not throw any error. Would you like to apply this approach also for other places?

Reply:
if we need this more than once (here), then I would suggest to do our `test_number_limits()` function so we can use it here e.g. with `lower = 0, include_lower = FALSE` arguments here